discord_bot2.py

```diff
@@ -67,6 +68,149 @@ def from_json(jobj: dict, thread: discord.Thread) -> 'Conversation':
         return Conversation(**jobj)
 
 
+async def display_help(message):
+    await message.channel.send(
+        "!restart - restart the bot\n"
+        "!log - print the log file\n"
+        "!rmlog - remove the log file\n"
+        "!status - print a status message\n"
+        "!help - print this message\n"
+    )
+
+
+async def execute_command(text, channel):
+    """
+    Execute a command in the shell and return the output to the channel
+    """
+    # Run command using shell and pipe output to channel
+    work_dir = Path(__file__).parent
+    await send(channel, f"```ps\n$ {text}```")
+    process = subprocess.run(text, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=work_dir)
+    # Get output of command and send to channel
+    errors = process.stderr.decode('utf-8')
+    if errors:
+        await send(channel, f'Errors: ```{errors}```')
+    output = process.stdout.decode('utf-8')
+    if output:
+        await send(channel, f'```{output}```')
+    return
+
+
+async def restart(message):
+    """
+    Restart the bot
+    :param message: The message that triggered the restart
+    """
+    await message.channel.send(f'Restart requested.')
+    await execute_command('git fetch', message.channel)
+    await execute_command('git reset --hard', message.channel)
+    await execute_command('git clean -f', message.channel)
+    await execute_command('git pull --rebase=false', message.channel)
+    await execute_command('poetry install', message.channel)
+    await message.channel.send(f'Restarting.')
+    subprocess.Popen(["bash", "restart.sh"])
+    return
+
+
+async def control_on_message(message):
+    """
+    This function is called whenever the bot sees a message in a control channel
+    :param message:
+    :return:
+    """
+    content = message.content
+    if content.startswith('!restart'):
+        await restart(message)
+
+    elif content.startswith('!log'):
+        await message.channel.send(file=discord.File('/tmp/duck.log'))
```

Comment:
Future PR: configure the Logger to write to the log file. Configure the log file path in the command-line args, then use that path here too.